src/opm/input/eclipse/Schedule/RFTConfig.cpp

```diff
@@ -21,196 +21,281 @@
+        // RFT data at well open is already recorded.  Don't trigger new RFT
+        // output event.
         return {};
+    }
 
     auto new_rft = *this;
-    for (auto iter = new_rft.rft_state.begin(); iter != new_rft.rft_state.end(); ) {
-        if (iter->second == RFT::YES)
-            iter = new_rft.rft_state.erase(iter);
-        else
-            iter++;
+
+    if (this->first_open_rft) {
+        // Well opens at this time and user requests RFT data on well open
+        // for all new wells.  Trigger RFT output.
+        new_rft.update(wname, RFT::YES);
+        new_rft.open_wells.insert_or_assign(wname, true);
+
+        return new_rft;
     }
 
-    for (auto iter = new_rft.plt_state.begin(); iter != new_rft.plt_state.end(); ) {
-        if (iter->second == PLT::YES)
-            iter = new_rft.plt_state.erase(iter);
-        else
-            iter++;
+    auto rft_fopn = new_rft.rft_state.find(wname);
+    if ((rft_fopn != new_rft.rft_state.end()) &&
+        (rft_fopn->second == RFT::FOPN))
+    {
+        // Well opens at this time and user requests RFT data on well open
+        // for this particular well.  Trigger RFT output.
+        rft_fopn->second = RFT::YES;
+    }
+
+    new_rft.open_wells.insert_or_assign(wname, true);
+
+    return new_rft;
+}
+
+std::optional<RFTConfig> RFTConfig::next() const
+{
+    // Next block configured by removing all 'YES' nodes from *this.  We do
+    // this because the 'YES' nodes have already triggered by the time the
+    // next block runs.
+    const auto rft_is_yes = [](const auto& rft_pair) {
+        return rft_pair.second == RFT::YES;
+    };
+
+    const auto plt_is_yes = [](const auto& plt_pair) {
+        return plt_pair.second == PLT::YES;
+    };
+
+    const auto rft_has_yes = mapContains(this->rft_state, rft_is_yes);
+    const auto plt_has_yes = mapContains(this->plt_state, plt_is_yes);
+
+    if (! (rft_has_yes || plt_has_yes)) {
+        // No 'YES' node in either the RFT or the PLT states.  Return
+        // nullopt to signify that next block is unchanged from current.
+        return {};
+    }
+
+    // Prune 'YES' nodes from both RFT and PLT states to form next block.
+    auto new_rft = std::optional<RFTConfig>{*this};
+
+    if (rft_has_yes) {
+        pruneFromMapIf(new_rft->rft_state, rft_is_yes);
+    }
+
+    if (plt_has_yes) {
+        pruneFromMapIf(new_rft->plt_state, plt_is_yes);
     }
 
     return new_rft;
 }
 
+bool RFTConfig::operator==(const RFTConfig& data) const
+{
+    return (this->first_open_rft == data.first_open_rft)
+        && (this->rft_state == data.rft_state)
+        && (this->plt_state == data.plt_state)
+        && (this->open_wells == data.open_wells);
+}
+
+RFTConfig RFTConfig::serializeObject()
+{
+    RFTConfig rft_config;
+    rft_config.first_open(true);
+
+    rft_config.update("P-1", RFT::FOPN);
+    rft_config.update("P-2", PLT::TIMESTEP);
+
+    rft_config.open_wells.emplace("I-1", true);
+
```

Comment:
totally opaque to me what this does and why. Might be me. Maybe add a comment?

Reply:
> totally opaque to me what this does and why.

Please note that the `serializeObject()` member function is "only" used to establish an object in a non-default state to enable testing the serialization code.  These statements "simply" record a number of requests to populate every internal table.  That way we get the best test coverage.

> Maybe add a comment?

Sure, I can do that.